README.md

```diff
@@ -42,14 +42,25 @@
 </a>
 
 DataFusion is an extensible query engine written in [Rust] that
-uses [Apache Arrow] as its in-memory format. DataFusion's target users are
+uses [Apache Arrow] as its in-memory format.
+
+The core DataFusion libraries in this repository are not designed to be an out-of-the-box tool for end users. However,
+the following subprojects offer packaged versions of DataFusion.
```

Comment:
I think we can make this wording more concise (this is the main intro where we get eyeballs), perhaps like:

```suggestion
The DataFusion libraries in this repository are used to build data centric system software. Some subprojects offer packaged versions of DataFusion for end users.
```